src/js/jsx/shared/Datalist.jsx

```diff
@@ -383,12 +391,14 @@ define(function (require, exports, module) {
                 
                 width += 20 * iconCount; // 20 pixels is the computed width + padding of an svg icon
 
+                React.findDOMNode(this.refs.autocomplete).style.left = width + "px";
```

Comment:
This is adjusting where to place the autocompletion word fragment, not changing the width of the input. It's in Datalist so that it can be up to date on what has been entered so far in the text box and update whenever that changes.